tests/checkers/unittest_unicode.py

```diff
@@ -0,0 +1,880 @@
+                end_line=3 + start_lines,
+                # node=module,
+                args=None,
+                confidence=pylint.interfaces.HIGH,
+                col_offset=22,
+                end_col_offset=23,
+            ),
+            pylint.testutils.MessageTest(
+                msg_id="invalid-character-sub",
+                line=4 + start_lines,
+                end_line=4 + start_lines,
+                # node=module,
+                args=None,
+                confidence=pylint.interfaces.HIGH,
+                col_offset=20,
+                end_col_offset=21,
+            ),
+            pylint.testutils.MessageTest(
+                msg_id="invalid-character-esc",
+                line=5 + start_lines,
+                end_line=5 + start_lines,
+                # node=module,
+                args=None,
+                confidence=pylint.interfaces.HIGH,
+                col_offset=20,
+                end_col_offset=21,
+            ),
+        ]
+        with self.assertAddsMessages(*expected):
+            self.checker.process_module(module)
+
+    @pytest.mark.parametrize(
+        "codec_and_msg, char, msg_id",
+        [
+            pytest.param(
+                codec_and_msg,
+                char_msg[0],
+                char_msg[1],
+                id=f"{char_msg[1]}_{codec_and_msg[0]}",
+            )
+            for codec_and_msg, char_msg in itertools.product(
+                CODEC_AND_MSG,
+                (
+                    ("\0", "invalid-character-nul"),
+                    ("\N{ZERO WIDTH SPACE}", "invalid-character-zero-width-space"),
+                ),
+            )
+        ],
+    )
+    def test_bad_chars_that_would_currently_crash_python(
+        self,
+        char: str,
+        msg_id: str,
+        tmp_path: Path,
+        codec_and_msg: Tuple[str, Tuple[pylint.testutils.MessageTest]],
+    ):
+        """Special test for a file containing chars that lead to
+
+        Python or Astroid crashes (which causes Pylint to exit early)
+        """
+        codec, start_msg = codec_and_msg
+        test_file = tmp_path / "failing.py"
+        test_file.write_text(
+            f"# file containing {char} will fail loading in AST checker \n"
+            f"# but we still want to check it, in case this behavior changes \n",
+            encoding=codec,
+        )
+        module = FakeNode(test_file)
+
+        expected = [
+            *start_msg,
+            pylint.testutils.MessageTest(
+                msg_id=msg_id,
+                line=1,
+                end_line=1,
+                # node=module,
+                args=None,
+                confidence=pylint.interfaces.HIGH,
+                col_offset=18,
+                end_col_offset=19,
+            ),
+        ]
+
+        with self.assertAddsMessages(*expected):
+            self.checker.process_module(cast(nodes.Module, module))
+
+
+class TestNoneAsciiChecker(pylint.testutils.CheckerTestCase):
```

Comment:
Same here, we can separate the unittest into two files.

Reply:
separated